pkg/ocm/sca/sca.go

```
@@ -180,18 +189,26 @@ func (c *Controller) requestDataAndCheckSecret(ctx context.Context, endpoint str
 	})
 }
 
-func (c *Controller) processResponses(ctx context.Context, responses Response) error {
+func (c *Controller) processResponses(ctx context.Context, responses Response, controlPlaneArch string) error {
 	if responses.Total == 1 {
 		// If there is only one architecture then we will use the secret name "etc-pki-entitlement"
 		// without the arch suffix to keep the backward compatibility
 		return c.checkSecret(ctx, &responses.Items[0], secretName)
 	}
 
-	for _, response := range responses.Items {
-		secretArchName := fmt.Sprintf(secretArchName, archMapping[response.Metadata.Arch])
+	controlPlaneCertData := responses.getCertDataByName(controlPlaneArch)
+	if controlPlaneCertData != nil {
+		// Create or update the default "etc-pki-entitlement" secret with the control plane node data
+		if err := c.checkSecret(ctx, controlPlaneCertData, secretName); err != nil {
+			return err
+		}
+	} else {
+		klog.Errorf("master node architecture not found, default secret is not created nor updated")
```

I think it would be better to return this error and handle it accordingly.